tests/benchdnn/graph/deserialize.cpp

```diff
@@ -516,6 +524,83 @@ const deserialized_op_t &deserialized_graph_t::get_op_by_in_lt(
     return dummy;
 }
 
+void deserialized_graph_t::detect_recognized_patterns() {
+    if (ops_.size() >= 3 && detect_sdpa_impl()) {
+        recognized_pattern_ = graph_recognized_pattern_t::sdpa;
+        return;
+    }
+}
+
+bool deserialized_graph_t::detect_sdpa_impl() const {
+
+    static const std::unordered_set<std::string> mm1_post_op_kind
+            = {"Divide", "Multiply", "Add", "Select", "GenIndex",
+                    "GreaterEqual", "StaticReshape", "StaticTranspose"};
+    const auto is_root_op = [&](const deserialized_op_t &op) {
+        return std::none_of(op.in_lts_.begin(), op.in_lts_.end(),
+                [&](const deserialized_lt_t &lt) {
+                    return !get_op_by_out_lt(lt.id_).empty();
+                });
+    };
+
+    std::vector<std::reference_wrapper<const deserialized_op_t>> starter_ops;
+    for (const auto &aop : ops_)
+        if (is_root_op(aop)) starter_ops.emplace_back(aop);
+
+    const auto find_next_util
+            = [&](const deserialized_op_t &start_op,
+                      const std::string &target_kind,
+                      const std::unordered_set<std::string> &allowed_skips)
+            -> deserialized_op_t {
+        auto current = start_op;
+        while (!current.empty() && current.kind_ != target_kind) {
+            if (!allowed_skips.empty() && allowed_skips.count(current.kind_)) {
+                break;
+            }
```

Comment:
Based on this code, current kind is not desired, allowed_skips contain that undesired kind, the loop breaks and the function returns empty op. What does it "allow to skip" then?
Would be also nice to see a documented intent and how to use the function.